src/utp_stream.cpp

```diff
@@ -799,6 +821,23 @@ void utp_socket_drained(utp_socket_impl* s)
 	s->maybe_trigger_send_callback();
 }
 
+boost::int32_t get_optimized_receive_buf_capacity(boost::uint32_t const rtt)
```

Comment:
the build error here is that you should either put this in the anonymous namespace (to get internal linkage) or add a forward declaration in the header. In order to have a unit test for it it has to have the forward declaration.